go/builder.go

```diff
@@ -324,6 +349,9 @@ func (b *Builder) CreateSharedString(s string) UOffsetT {
 
 // CreateString writes a null-terminated string as a vector.
 func (b *Builder) CreateString(s string) UOffsetT {
+	if len(s) == 0 {
```

Comment:
Is this required?

Reply:
old one
```
// CreateString writes a null-terminated string as a vector.
func (b *Builder) CreateString(s string) UOffsetT {
	b.assertNotNested()
	b.nested = true

	b.Prep(int(SizeUOffsetT), (len(s)+1)*SizeByte)
	b.PlaceByte(0)

	l := UOffsetT(len(s))

	b.head -= l
	copy(b.Bytes[b.head:b.head+l], s)

	return b.EndVector(len(s))
}
```

add bypass code
```
func (b *Builder) CreateString(s string) UOffsetT {
	if len(s) == 0 {
		return UOffsetT(0)
	}
	b.assertNotNested()
	b.nested = true

	b.Prep(int(SizeUOffsetT), (len(s)+1)*SizeByte)
	b.PlaceByte(0)

	l := UOffsetT(len(s))

	b.head -= l
	copy(b.Bytes[b.head:b.head+l], s)

	return b.EndVector(len(s))
}
```


Consider creating an fbs builder at runtime, where the data fields come from user input,  network, etc. builders should be fast / robust / proper error (exception) handling.  for 0-length  string / binary array ... should return obvious result and reduce reduce unnecessary CPU operations. 

In fact, for a 0-length string, there is no bypass code snippet, the result is the same after processing ( result is  0 ).